extensions/functions_datetime.yaml

```diff
@@ -3,36 +3,111 @@
+      * SECOND Return the second (0-59).
+      * MILLISECOND Return number of milliseconds since the last full second.
+      * MICROSECOND Return number of microseconds since the last full millisecond.
+      * SUBSECOND Return number of microseconds since the last full second of the given timestamp.
+      * UNIX_TIME Return number of seconds that have elapsed since 1970-01-01 00:00:00 UTC, ignoring leap seconds.
+      * TIMEZONE_OFFSET Return number of seconds of timezone offset to UTC.
+
+      Timezone strings must be as defined by IANA timezone database (https://www.iana.org/time-zones).
+      Examples: "Pacific/Marquesas", "Etc/GMT+1".
+      If timezone is invalid an error is thrown.
     impls:
       - args:
           - name: component
-            options: [ YEAR, MONTH, DAY, SECOND ]
+            options: [ YEAR, ISO_YEAR, US_YEAR, QUARTER, MONTH, DAY, DAY_OF_YEAR, MONDAY_DAY_OF_WEEK, SUNDAY_DAY_OF_WEEK,
+                       MONDAY_WEEK, SUNDAY_WEEK, ISO_WEEK, US_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND,
+                       SUBSECOND, UNIX_TIME, TIMEZONE_OFFSET ]
             description: The part of the value to extract.
           - name: x
-            value: timestamp
+            value: timestamp_tz
+          - name: timezone
+            description: Timezone string from IANA tzdb.
+            value: string
         return: i64
       - args:
           - name: component
-            options: [ YEAR, MONTH, DAY, SECOND ]
+            options: [ YEAR, ISO_YEAR, US_YEAR, QUARTER, MONTH, DAY, DAY_OF_YEAR, MONDAY_DAY_OF_WEEK, SUNDAY_DAY_OF_WEEK,
+                       MONDAY_WEEK, SUNDAY_WEEK, ISO_WEEK, US_WEEK, HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND,
+                       SUBSECOND, UNIX_TIME ]
             description: The part of the value to extract.
           - name: x
-            value: timestamp_tz
+            value: timestamp
         return: i64
       - args:
           - name: component
-            options: [ YEAR, MONTH, DAY ]
+            options: [ YEAR, ISO_YEAR, US_YEAR, QUARTER, MONTH, DAY, DAY_OF_YEAR, MONDAY_DAY_OF_WEEK, SUNDAY_DAY_OF_WEEK,
+                       MONDAY_WEEK, SUNDAY_WEEK, ISO_WEEK, US_WEEK, UNIX_TIME ]
             description: The part of the value to extract.
           - name: x
             value: date
         return: i64
       - args:
           - name: component
-            options: [ SECOND ]
+            options: [ HOUR, MINUTE, SECOND, MILLISECOND, MICROSECOND, SUBSECOND ]
             description: The part of the value to extract.
           - name: x
             value: time
         return: i64
+  -
+    name: "extract_boolean"
+    description: >-
+      Extract boolean values of a date/time value.
+      * IS_LEAP_YEAR Return true if year of the given timestamp is a leap year and false otherwise.
+      * IS_DST Return true if DST (Daylight Savings Time) is observed at the given timestamp
+        in the given timezone.
```

Comment:
```suggestion
      * IS_LEAP_YEAR Return true if year of the given value is a leap year and false otherwise.
      * IS_DST Return true if DST (Daylight Savings Time) is observed at the given value
        in the given timezone.
```